src/Services/CDADocumentService.php

```diff
@@ -173,4 +197,40 @@ public function portalGenerateCCDZip($pid): string
 
         return $response->getContent();
     }
+
+    /**
+     * Complete zip of xml, html version
+     * when called within an openemr authorized session.
+     *
+     * @param $pid
+     * @return string
+     */
+    public function generateCCDZip($pid): string
+    {
+        $parameterArray = array(
+            'combination' => $pid,
+            'components' => 'allergies|medications|problems|immunizations|procedures|results|plan_of_care|vitals|social_history|encounters|functional_status|referral|instructions|medical_devices|goals',
+            'downloadccda' => 'download_ccda',
+            'latestccda' => '0',
+            'send_to' => 'download_all',
+            'sent_by_app' => 'core_api',
+            'ccda_pid' => [0 => $pid],
+            'view' => 0,
+            'recipient' => 'self',
+            'site' => $_SESSION ['site_id'],
```

Comment:
i think there is a space between $_SESSION and the `[` bracket

Reply:
hmm wonder if that would still work